privcount/data_collector.py

```diff
@@ -139,7 +139,30 @@ def do_start(self, config): # called by protocol
             logging.info('refusing to start collecting without required share keepers')
             return None
 
-        self.aggregator = Aggregator(dc_counters, config['sharekeepers'], self.config['noise_weight'], counter_modulus(), self.config['event_source'])
+        # validate the traffic model we got from the tally server
+        traffic_model_valid = False
+        if 'traffic_model' in config:
+            traffic_model_valid = True
+            for k in ['states', 'emission_probability', 'transition_probability', 'start_probability']:
+                if k not in config['traffic_model']:
+                    traffic_model_valid = False
+        # load the traffic model object that we will use during aggregation
+        tmodel = None
+        if traffic_model_valid:
+            # build the model
+            states = config['traffic_model']['states']
+            start_p = config['traffic_model']['start_probability']
+            trans_p = config['traffic_model']['transition_probability']
+            emit_p = config['traffic_model']['emission_probability']
+            tmodel = TrafficModel(states, start_p, trans_p, emit_p)
+
+            # now add in the keys needed for counting
+            # NOTE the tally server send these, but we ignored them above
+            for label in tmodel.get_counter_labels():
+                # XXX TODO FIXME these sigmas should not be 0
+                dc_counters[label] = {'bins': [[0.0, float("inf")]], 'sigma': 0.0}
```

Comment:
I don't understand where the sigmas come from for the traffic model, so I can't work out what this TODO means.